svf/include/Graphs/BasicBlockG.h

```diff
@@ -0,0 +1,290 @@
+        pred->addOutgoingEdge(edge);
+        this->predBBs.push_back(pred);
+        pred->succBBs.push_back(this);
+    }
+
+    inline const SVFFunction* getParent() const
+    {
+        return fun;
+    }
+
+    inline const SVFFunction* getFunction() const
+    {
+        return fun;
+    }
+
+    inline const ICFGNode* front() const
+    {
+        assert(!allICFGNodes.empty() && "bb empty?");
+        return allICFGNodes.front();
+    }
+
+    inline const ICFGNode* back() const
+    {
+        assert(!allICFGNodes.empty() && "bb empty?");
+        return allICFGNodes.back();
+    }
+
+    inline std::vector<const SVFBasicBlock*> getSuccessors() const
+    {
+        std::vector<const SVFBasicBlock*> res;
+        for (auto edge : this->getOutEdges())
+        {
+            res.push_back(edge->getDstNode());
+        }
+        return res;
+    }
+
+    inline std::vector<const SVFBasicBlock*> getPredecessors() const
+    {
+        std::vector<const SVFBasicBlock*> res;
+        for (auto edge : this->getInEdges())
+        {
+            res.push_back(edge->getSrcNode());
+        }
+        return res;
+    }
+    u32_t getNumSuccessors() const
+    {
+        return this->getOutEdges().size();
+    }
+    u32_t getBBSuccessorPos(const SVFBasicBlock* Succ) {
+        u32_t i = 0;
+        for (const SVFBasicBlock* SuccBB: succBBs)
+        {
+            if (SuccBB == Succ)
+                return i;
+            i++;
+        }
+        assert(false && "Didn't find successor edge?");
+        return 0;
+    }
+    u32_t getBBSuccessorPos(const SVFBasicBlock* Succ) const {
+        u32_t i = 0;
+        for (const SVFBasicBlock* SuccBB: succBBs)
+        {
+            if (SuccBB == Succ)
+                return i;
+            i++;
+        }
+        assert(false && "Didn't find successor edge?");
+        return 0;
+
+    }
+    u32_t getBBPredecessorPos(const SVFBasicBlock* succbb) {
+        u32_t pos = 0;
+        for (const SVFBasicBlock* PredBB : succbb->getPredecessors())
+        {
+            if(PredBB == this)
+                return pos;
+            ++pos;
+        }
+        assert(false && "Didn't find predecessor edge?");
+        return pos;
+    }
+    u32_t getBBPredecessorPos(const SVFBasicBlock* succbb) const {
+        u32_t pos = 0;
+        for (const SVFBasicBlock* PredBB : succbb->getPredecessors())
+        {
+            if(PredBB == this)
+                return pos;
+            ++pos;
+        }
+        assert(false && "Didn't find predecessor edge?");
+        return pos;
+    }
+
+    const std::string& getName() const override;
+    const std::string toString() const;
+
+};
+
+
+
+typedef GenericGraph<SVFBasicBlock, BasicBlockEdge> GenericBasicBlockGraphTy;
+class BasicBlockGraph: public GenericBasicBlockGraphTy
+{
+private:
+    NodeID id{0};
+public:
+    /// Constructor
+    BasicBlockGraph() {
+
+    }
+
+    SVFBasicBlock* addBasicBlock(const SVFFunction* f)
```

Comment:
SVFFunction is a BasicBlock? It looks strange here.

Reply:
I'll rename it